test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,318 @@
+	// List the pods, making sure we observe all the replicas.
+	listTimeout := time.Minute
+	listInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	var pods *api.PodList
+	t := time.Now()
+	for {
+		var err error
+		pods, err = c.Pods(ns).List(label, fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+
+		Logf("Controller %s: Found %d pods out of %d", name, len(pods.Items), replicas)
+		if len(pods.Items) == replicas {
+			break
+		}
+		if time.Since(t) > listTimeout {
+			Failf("Controller %s: Gave up waiting for %d pods to come up after seeing only %d pods after %v seconds",
+				name, replicas, len(pods.Items), time.Since(t).Seconds())
+		}
+		time.Sleep(listInterval)
+	}
+	return pods
+}
+
+func waitForPodsRunning(c *client.Client, pods *api.PodList) {
+	// Wait for the pods to enter the running state. Waiting loops until the pods
+	// are running so non-running pods cause a timeout for this test.
+	By("ensuring each pod is running")
+	for _, pod := range pods.Items {
+		err := waitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func verifyPodsListining(c *client.Client, ns, name string, pods *api.PodList) {
+	By("trying to dial each unique pod")
+	retryTimeout := 2 * time.Minute
+	retryInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	err := wait.Poll(retryInterval, retryTimeout, podResponseChecker{c, ns, label, name, pods}.checkAllResponses)
+	if err != nil {
+		Failf("Did not get expected responses within the timeout period of %.2f seconds: %v", retryTimeout.Seconds(), err)
+	}
+}
+
+var _ = Describe("ResizeNodes", func() {
+	var c *client.Client
+	var ns string
+	var testingNs *api.Namespace
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err = createTestingNS("resize-nodes", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !providerIs("gce") {
+			return
+		}
+		By(fmt.Sprintf("destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+		By(fmt.Sprintf("restoring the original node instance group size"))
+		if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original cluster size: %s", err)
+		}
+	})
+
+	It("should be able to delete nodes.", func() {
+		Logf("starting test %v", testContext.CloudConfig.NumNodes)
+
+		if !providerIs("gce") {
```

Comment:
make sure this is consistent with what you have above. Could you please refactor this and the above to use the same var so it's easier to keep them up to date? If, for example, we turn them on above, but not here, we'll think the tests are passing even though the bulk is being skipped.